### PR TITLE
feat: add support for Azure OpenAI

### PR DESCRIPTION
**Pull request by quitrk**

### CODE REVIEW

File: credentials.yaml.sample [hunk 1]
```diff
@@ -1,4 +1,19 @@
...
+            secret: test_secret
+            type: AZURE_OPENAI
+        OPENAI:
+            customerId: testCustomerId
+            enabled: false
+            metadata:
+                model: gpt-3
+            secret: test_secret
+            type: OPENAI
+    customerId: testCustomerId
```
Comment (saghul) [inline credentials.yaml.sample:19]: stray line?

File: skynet/auth/openai.py [hunk 2]
```diff
@@ -40,4 +47,13 @@ async def setup_credentials():
 
 
 def get_credentials(customer_id):
-    return credentials.get(customer_id, {}) or {}
+    customer_credentials = credentials.get(customer_id, {}) or {}
+    multiple_credentials = customer_credentials.get('credentialsMap')
+
+    if multiple_credentials:
+        result = [val for val in multiple_credentials.values() if val['enabled'] == True]
```
Comment (saghul) [inline skynet/auth/openai.py:54]:
```suggestion
        result = [val for val in multiple_credentials.values() if val['enabled']]
```

Comment (saghul) [reply skynet/auth/openai.py:54]: No need to check explicitly for `True`, unless that thing can have any other truthy values?

**Review by saghul — APPROVED** [review]
2 minor comments, great work!

File: credentials.yaml.sample (hunk 1 above)
Comment (quitrk) [reply credentials.yaml.sample:19]: wdym? it's an actual field in the yaml

Comment (quitrk) [reply credentials.yaml.sample:19]: though we don't really use it so we could get rid of it for the sample

**Pull request merged**